Merge from onprc19.1 & onprc19.1Prod r.63271 to 65949

labkey-bpatel opened the pull request:
#### Rationale
Merge changes from onprc19.1 r.63271 to 65757
Merge changes from onprc19.1Prod r. 65762 to 65949

#### Related Pull Requests
https://github.com/LabKey/platform/pull/1626
https://github.com/LabKey/ehrModules/pull/74
https://github.com/LabKey/DiscvrLabKeyModules/pull/64
https://github.com/LabKey/LabDevKitModules/pull/55
https://github.com/LabKey/sampleManagement/pull/381
https://github.com/LabKey/biologics/pull/712
https://github.com/LabKey/labbook/pull/53
https://github.com/LabKey/wnprc-modules/pull/38

labkey-bpatel committed:
Merge from onprc19.1 r.63271

labkey-bpatel committed:
Merge from onprc19.1 r.63271 (includes newly created folders & files that didn't find the new base automatically)

labkey-bpatel committed:
Merge from onprc19.1 r.63282

labkey-bpatel committed:
Merge from onprc19.1 r.63282

labkey-bpatel committed:
Merge from onprc19.1 r.63328 to 64381

labkey-bpatel committed:
Merge from onprc19.1 r.64647

labkey-bpatel committed:
Merge from onprc19.1 r.64748

labkey-bpatel committed:
Merge from onprc19.1 r.64769 to 64894

labkey-bpatel committed:
Merge from onprc19.1 r.64939 to 64951

labkey-bpatel committed:
Merge from onprc19.1 r.64953 to 65052

labkey-bpatel committed:
Merge from onprc19.1 r.65053 to 65061

labkey-bpatel committed:
Merge from onprc19.1 r.65063, 65064, 65071, 65388 to 65398

labkey-bpatel committed:
Merge from onprc19.1 r.65410

labkey-bpatel committed:
Merge from onprc19.1 r.65412 to 65638

labkey-bpatel committed:
Merge from onprc19.1 r.65639 to 65677

labkey-bpatel committed:
Merge from onprc19.1 r.65678 to 65757

labkey-bpatel committed:
Merge from onprc19.1 r.64748 -- include missed .gif file

labkey-bpatel committed:
Fix build issues

labkey-bpatel committed:
Merge from onprc19.1Prod r 65762 to 65949

labkey-bpatel committed:
Merge from onprc19.1Prod r 65762 to 65949 - include missed changes

labkey-bpatel committed:
Add LDK module dependency

labkey-jeckels commented on onprc_ehr/resources/queries/onprc_ehr/vet_assignment_summary.sql:
```diff
@@ -13,11 +13,33 @@
  * limitations under the License.
  */
+/*
+ * Copyright (c) 2014 LabKey Corporation
```
Double-licensed.

labkey-bpatel replied:
dupe removed

labkey-jeckels commented on onprc_ehr/resources/queries/study/Clinical Remarks.query..xml:
```diff
@@ -1,15 +1,18 @@
 <query xmlns="http://labkey.org/data/xml/query">
```
File has extra ".." in its name

labkey-bpatel replied:
yikes, thanks for catching that! now fixed

labkey-jeckels commented on onprc_ehr/resources/queries/study/StudyDetails_cohort_values.query.xml:
```diff
@@ -0,0 +1,18 @@
+                        <isHidden>false</isHidden>
+                    </column>
+                </columns>
+                <sorts>
```
<sorts> is illegal in .query.xml files

labkey-bpatel replied:
<sorts> now removed

labkey-jeckels commented on onprc_ehr/resources/queries/study/StudyDetails_group_values.query.xml:
```diff
@@ -0,0 +1,18 @@
+                        <isHidden>false</isHidden>
+                    </column>
+                </columns>
+                <sorts>
```
Illegal here too

labkey-bpatel replied:
<sorts> now removed

labkey-jeckels commented on onprc_ehr/resources/queries/study/StudyDetails_phase_values.query.xml:
```diff
@@ -0,0 +1,18 @@
+                        <isHidden>false</isHidden>
+                    </column>
+                </columns>
+                <sorts>
```
Still illegal!

labkey-bpatel replied:
lot of illegal business going on, now removed

labkey-jeckels commented on onprc_ehr/resources/schemas/dbscripts/sqlserver/onprc_ehr-12.395-12.396.sql:
```diff
@@ -0,0 +1,126 @@
+ */
+
+/****** Object:  StoredProcedure [onprc_ehr].[etl1_eIACUCtoPRIMEProcessing]    Script Date: 2/7/2018 2:01:46 PM ******/
+SET ANSI_NULLS ON
```
We should really strip out all of these SET ANSI_NULLS ON and SET QUOTED_IDENTIFIER ON statements out of these upgrade scripts. I won't comment on each one individually.

labkey-bpatel replied:
done

labkey-jeckels commented on onprc_ehr/resources/schemas/dbscripts/sqlserver/onprc_ehr-12.395-12.396.sql:
```diff
@@ -0,0 +1,126 @@
+/*
```
Since we're already at 18.10 in the 20.7 branch we'll have to merge these scripts manually.

labkey-bpatel replied:
onprc_ehr and onprc_billing scripts are now rolled up:
onprc_ehr upgrade scripts (12.20 to 17.705) consolidated to onprc_ehr-0.00-18.10.
onprc_billing scripts (12.30 to 17.502) consolidated to onprc_billing-0.00-18.10.

labkey-jeckels replied:
Sorry I wasn't more specific. I wasn't worried about rolling up the scripts. I'm worried that devs on the 20.7 branch will already have the onprcEhrModules checked out, and they'll be on version 18.10 of the schema. Therefore, they'll never run these scripts, whether they're incrementals or rollups.

labkey-jeckels commented on onprc_ehr/resources/web/onprc_ehr/window/MarkAssignmentCompletedWindow.js:
```diff
@@ -43,7 +43,7 @@ Ext4.define('ONPRC_EHR.window.MarkAssignmentCompletedWindow', {
             bodyStyle: 'padding: 5px;',
             items: [{
-                html: 'This helper allows you to end the selected assignments.  You are required to choose the end date, and release type.  If release condition is left blank, the projected release condition will be used.',
+                html: 'This helperT allows you to end the selected assignments.  You are required to choose the end date, and release type.  If release condition is left blank, the projected release condition will be used.',
```
Typo!

labkey-bpatel replied:
fixed

labkey-jeckels commented on onprc_ehr/src/org/labkey/onprc_ehr/history/ONPRCClinicalRemarksDataSource.java:
```diff
@@ -0,0 +1,118 @@
+        if (rs.hasColumn(FieldKey.fromString(field)) && rs.getObject(FieldKey.fromString(field)) != null)
+        {
+            sb.append("<tr style='vertical-align:top;margin-bottom: 5px;'><td style='padding-right: 5px;'>" + label + ":</td><td>");
+            sb.append(rs.getString(FieldKey.fromString(field)));
```
Should be HTML-encoded

labkey-bpatel replied:
done

labkey-jeckels commented on onprc_ehr/src/org/labkey/onprc_ehr/notification/InfantsBornAssignedNotification.java:
```diff
@@ -0,0 +1,172 @@
+                    String dams = results.getString(FieldKey.fromString("dam"));
+                    String projectname = results.getString(FieldKey.fromString("ProjectName"));
+
+                    msg.append("<tr><td>" + Ids  + "</td><td>" + getDateTimeFormat(c).format(datess) + "</td><td>"  + projectname + "</td><td>" + dams + "</td></tr>\n");
```
HTML encoding

labkey-bpatel replied:
done

labkey-jeckels commented on onprc_ehr/src/org/labkey/onprc_ehr/notification/ObeseFlagNotification.java:
```diff
@@ -156,7 +165,20 @@ public void exec(ResultSet rs) throws SQLException
+                    msg.append("<td>" + Ids  + "</td>" +
+                            "<td>" + rooms +  "</td><td>"  + cages  + "</td>" +
+                            "<td>" + "  " + weights  + "</td><td>" +
+                            DateUtil.formatDateTime(c, datess) + "</td><td>" + DateUtil.formatDateTime(c, enddates) + "</td><td>" + "  " + vetname  + "</td></tr>\n");
```
HTML encoding

labkey-bpatel replied:
done

labkey-jeckels commented on onprc_ehr/src/org/labkey/onprc_ehr/notification/TreatmentAlertsPostOpsNotification.java:
```diff
@@ -176,13 +175,111 @@ public void exec(ResultSet object) throws SQLException
+                            msg.append("<tr bgcolor = " + '"' + "#FFFF00" + '"' + ">");
+                        }
+
+                        msg.append("<td>" + rs.getString("id") + "</td>");
```
HTML encoding

labkey-bpatel replied:
done

labkey-jeckels commented on onprc_ehr/src/org/labkey/onprc_ehr/notification/TreatmentAlertsPostOpsNotificationSecondary.java:
```diff
@@ -171,6 +171,104 @@ public void exec(ResultSet object) throws SQLException
+                            msg.append("<tr bgcolor = " + '"' + "#FFFF00" + '"' + ">");
+                        }
+
+                        msg.append("<td>" + rs.getString("id") + "</td>");
```
HTML encoding

labkey-bpatel replied:
done

labkey-jeckels commented on onprc_ehr/src/org/labkey/onprc_ehr/notification/TreatmentAlertsPostOpsNotificationThird.java:
```diff
@@ -0,0 +1,287 @@
+                            msg.append("<tr bgcolor = " + '"' + "#FFFF00" + '"' + ">");
+                        }
+
+                        msg.append("<td>" + rs.getString("id") + "</td>");
```
HTML encoding

labkey-bpatel replied:
done

labkey-jeckels commented on onprc_ehr/src/org/labkey/onprc_ehr/notification/UnoccupiedRoomsNotification.java:
```diff
@@ -122,4 +143,62 @@ public void exec(ResultSet rs) throws SQLException
+
+                {
+                    Results rs = new ResultsImpl(object, colMap);
+                    msg.append("<tr><td>" + (rs.getString("building") == null ? "" : rs.getString("building")) + "</td><td>" + rs.getString("room") + "</td></tr>");
```
HTML encoding

labkey-bpatel replied:
done

labkey-jeckels commented on onprc_ehr/src/org/labkey/onprc_ehr/query/ONPRC_EHRTriggerHelper.java:
```diff
@@ -2071,44 +2300,135 @@ public void sendProtocolNotifications(String protocolid)
+                               {
+                                   for (String Investname : ret2)
+                                   {
+                                       html.append("<tr><td>" + (rs.getString("external_id") == null ? "" : rs.getString("external_id")) + "</td><td>  " + Investname + "   </td><td>" + rs.getString("approve") + "</td></tr>\n");
```
HTML encoding

labkey-bpatel replied:
done

labkey-jeckels commented on onprc_ehr/src/org/labkey/onprc_ehr/query/ONPRC_EHRTriggerHelper.java:
```diff
@@ -2071,44 +2300,135 @@ public void sendProtocolNotifications(String protocolid)
+                               {
+                                   for (String Investname : ret2)
+                                   {
+                                       html.append("<tr><td>" + ( rs.getString("name"))  + "</td><td>" + ( rs.getString("project"))  + "</td><td>   " + ( rs.getString(protocolColumn.getAlias()))  + "</td><td>   " + Investname + "   </td><td>" + rs.getString("startdate") + "</td></tr>\n");
```
HTML encoding

labkey-bpatel replied:
done

labkey-jeckels reviewed (changes requested):
A few merge-related problems and a handful of other fixes I noticed we should make during the review.

labkey-bpatel committed:
Add database tables to the xml metadata

labkey-bpatel committed:
Add database tables to the xml metadata

labkey-bpatel committed:
Code review changes
- Cleanup in sql scripts
- Add HTML encoding
- Remove illegal usage of <sort> in .query.xml files

labkey-bpatel committed:
Consolidate onprc_ehr upgrade scripts (12.20 to 17.705) to onprc_ehr-0.00-18.10.
Consolidate onprc_billing scripts (12.30 to 17.502) to onprc_billing-0.00-18.10.

labkey-bpatel committed:
Code review change - Register MarkCompletedButton for 'geneticAncestry' (instead of registering in EHR module).

labkey-jeckels committed:
More SQL script cleanup

labkey-jeckels reviewed (changes requested):
All looks good except there's more to do on the upgrade scripts

labkey-bpatel committed:
Include 'GO' before 'Create Procedure' statements to avoid 'CREATE/ALTER PROCEDURE' must be the first statement in a query batch' errors.

labkey-bpatel committed:
Handle observation_types deletion for dbs on different versions.

labkey-bpatel commented:
Only script that needed to be handled was the one that deletes observation_types.
With recent changes, 20.7 server (& onprc19.1Prod) can upgrade to this fb without issues.
Let me know if i am missing anything else.

labkey-bpatel committed:
Include contents of upgrade scripts from onprc19.1Prod that are not in release20.7-SNAPSHOT

labkey-bpatel committed:
Substitute module property path. Fix query name (demograophis to demographics).

labkey-bpatel committed:
Add the contents of RateCalc.sql to the upgrade script. Remove RateCalc.sql.

labkey-bpatel committed:
-Replace hardcoded path with moduleProperty in sql queries.
-Add query stored in the db to the module.
-Remove broken query onprc_billing.leasefee_leasetypeTest.
-Add module property BillingContainer_Public.

labkey-bpatel committed:
-Replace hardcoded path with moduleProperty in sql queries.
-Remove broken query study.currentBloodAvailable.
-Add module property DCM_NHP_Resources_Container.
-Update automated tests.

labkey-bpatel committed:
Use getColumnName() instead of getSelectName() (otherwise it errors when viewing most of study datasets and also causes test failures).

labkey-bpatel committed:
Fix error caused by an 'undefined' value.

labkey-bpatel committed:
Fix typo, other minor cleanup & updates.

labkey-bpatel committed:
Automated test fix for testArrivalApi(). (Fix for testBloodVolumeApi() wip)

labkey-bpatel committed:
Add values to Surgical Repairs template. Move skipStudyImportQueryValidation() to the parent class.

labkey-bpatel committed:
Revert 'Add values to Surgical Repair template'.
Use existing & functional utility code to get ehr_lookups.labwork_services store.
Replace 'onprc-snomedtreatmentcombo' with 'ehr-snomedtreatmentcombo' (since the onprc one is not defined).
Some cleanup.

labkey-bpatel committed:
Test fixes. Associate 'code' column with editorConfig for ehr_lookups.drug_defaults.

labkey-bpatel committed:
Remove unnecessary code that was restricting getting drug formulary values/populating Frequency & Route values upon Treatment selection on Treatment Orders data entry form.

labkey-bpatel committed:
Remove version property

labkey-bpatel committed:
Get QC State by label instead of hardcoded rowid value.

labkey-bpatel committed:
Replace hardcoded value with module property. Add columns to demographics and pairings datasets.

labkey-bpatel committed:
Remove USE [Labkey]

labkey-jeckels committed:
Merge remote-tracking branch 'origin/20.7_fb_merge_from_onprc19.1' into 20.7_fb_merge_from_onprc19.1

# Conflicts:
#	onprc_billing/resources/queries/onprc_billing/RateCalc.sql
#	onprc_billing/resources/schemas/dbscripts/sqlserver/onprc_billing-17.501-17.502.sql
#	onprc_ehr/resources/schemas/dbscripts/sqlserver/onprc_ehr-12.395-12.396.sql
#	onprc_ehr/resources/schemas/dbscripts/sqlserver/onprc_ehr-12.405-12.406.sql
#	onprc_ehr/resources/schemas/dbscripts/sqlserver/onprc_ehr-12.406-12.407.sql
#	onprc_ehr/resources/schemas/dbscripts/sqlserver/onprc_ehr-12.407-12.408.sql
#	onprc_ehr/resources/schemas/dbscripts/sqlserver/onprc_ehr-12.408-12.409.sql
#	onprc_ehr/resources/schemas/dbscripts/sqlserver/onprc_ehr-17.702-17.703.sql
#	onprc_ehr/resources/schemas/dbscripts/sqlserver/onprc_ehr-20.106-20.107.sql

labkey-bpatel committed:
Remove skipStudyImportQueryValidation(), and other related refactor. Query updates (note: these queries still fail the validation, they are just partially updated/fixed).

labkey-jeckels committed:
Merge remote-tracking branch 'origin/20.7_fb_merge_from_onprc19.1' into 20.7_fb_merge_from_onprc19.1

labkey-jeckels committed:
Clean up long-defunct SQL queries

labkey-jeckels committed:
Clean up queries - delete defunct queries, add missing lists and columns, fix up XML
